src/grid/grid.component.ts

```diff
@@ -720,6 +720,9 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
     public pinColumn(columnName: string): boolean {
         const col = this.getColumnByName(columnName);
 
+        if (col.pinned) {
+            return;
```

Comment:
It should return false when no pinning is done. Also for the if can be combined with the if below.